modules/primo/src/Primo/Ting/Result.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * @file
+ * The Result class.
+ */
+
+namespace Primo\Ting;
+
+use Primo\BriefSearch\Document;
+use Ting\Search\TingSearchRequest;
+use Ting\Search\TingSearchResultInterface;
+use TingCollection;
+
+/**
+ * A Ting compatible search result from the Primo search provider.
+ *
+ * @package Primo\Ting
+ */
+class Result implements TingSearchResultInterface {
+
+  /**
+   * The original search request that was executed to produce this result.
+   *
+   * @var \Ting\Search\TingSearchRequest
+   */
+  protected $tingSearchRequest;
+
+  /**
+   * The primo-specific search result object.
+   *
+   * @var \Primo\BriefSearch\Result
+   */
+  protected $result;
+
+  /**
+   * List of Ting collections in the search result.
+   *
+   * As Primo does not support the concept of a collection we have a 1-1
+   * relation between materials and collections.
+   *
+   * @var \TingCollection[]
+   */
+  protected $collections;
+
+  /**
+   * Result constructor.
+   *
+   * @param \Primo\BriefSearch\Result $result
+   *   Primo search result.
+   *
+   * @param \Ting\Search\TingSearchRequest $ting_search_request
+   *   Ting search query request that was executed to produce the result.
+   */
+  public function __construct(\Primo\BriefSearch\Result $result, TingSearchRequest $ting_search_request) {
+    $this->result = $result;
+    $this->tingSearchRequest = $ting_search_request;
+
+    // Extract the document ids and load a collection collection pr. document.
+    $ids = array_map(function(Document $document) {
+      return $document->getRecordId();
+    }, $result->getDocuments());
+    $this->collections = entity_load('ting_collection', array(),
+      array('ding_entity_id' => $ids));
```

Comment:
Perhaps a style thing but I was thinking whether it would be more appropriate to move this logic to the getter. We do not use the `$collections` property for anything else and we should have all the information available in the other properties at that point anyway. 

On the other hand collection loading might be slow and thus it makes sense only to do it once. 🤔

Reply:
I've moved it down into the getter - does also have the advantage that it is possible to instantiate the class without involving core.